File: src/jEvropa2/data/Show.java
package jEvropa2.data;


import java.net.URL;

/**
 *
 * @author Skup5
 */
public class Show extends E2Data {

    private URL webSiteUrl;

    public Show(String name, URL webSiteUrl) {
        super(name);
        this.webSiteUrl = webSiteUrl;
    }

    public URL getWebSiteUrl() {
        return webSiteUrl;
    }

    public String info() {
        return getName() + " (" + getWebSiteUrl().toExternalForm() + ")";
    }
}

File: src/main/kt/cz/skup5/jEvropa2/data/Show.kt
package cz.skup5.jEvropa2.data

import java.net.URI

/**
 * Represents a category of [Item]s.
 *
 * @author Skup5
 */
data class Show(
        /** The show unique id. */
        override val id: Int = ID_NONE,
        /** The show title/label. */
        override val name: String,
        /** The absolute url of show web site. */
        var webSiteUri: URI,
        var slug: String
) : E2Data(id, name) {

    fun hasWebSiteUri(): Boolean = webSiteUri !== EMPTY_URI

    fun info(): String {
        return "$name ($id${if (hasWebSiteUri()) ", $webSiteUri" else ""}${if (slug.isNotEmpty()) ", $slug" else ""})"
    }
}
